tools/wpt/run.py

```diff
@@ -158,6 +161,62 @@ def check_environ(product):
                 raise WptrunError(message)
 
 
+class AndroidLogcat:
```

Comment:
My impression is that running logcat this way will collect a lot of logs, and would be difficult to scan through.

We previously have an implementation for Chromium Android browsers, which is trying to associate the log with the test run. We are not sure how that is useful either.

Reply:
Yeah, it collects the full log, which can indeed be large. But we write it to a separate artifact rather than intermixing it with the wptrunner logs. 

I think it makes more sense to start with a more complete log capture solution, and layer on filters later if we need them (FWIW gecko CI does not have filters, and basically uses the approach here).

Reply:
Sure, let's do this first.